MI-121: Add Image File Upload with Preview and Drag-n-Drop Support

kevinpjones opened the pull request:
## JIRA

[MI-121](https://lobsters.atlassian.net/browse/MI-121)

[Figma](https://www.figma.com/file/qz0wJ4yY4RGPy1JqPqCIsM/Campaigns-v2?type=design&node-id=1691-10662&mode=design&t=XzEIzfC5fQUAostj-0)
![image](https://github.com/lob/ui-components/assets/3537215/da5a388d-a93d-43e3-b7c3-61e5814b218c)
 
<!--
## Technical Design Doc
- link to TDD if the feature had one (optional)
-->

## Description

- Adds an image upload component with drag and drop support

<!--
## Screenshots
- before and after screenshots for features with visual changes
-->

![image](https://github.com/lob/ui-components/assets/3537215/e1f3e0ed-75ee-44c8-a365-ae9fedc05af3)

![image](https://github.com/lob/ui-components/assets/3537215/9beabbc3-7100-449a-ac51-022495ddc0b7)

<!--
## Tests
- info about what was used to validate the feature (more for larger, more complicated features)
-->

<!--
## Areas of Concern
- to call out what could be a problem (optional)
-->

<!--
## Questions
- to ask any questions for the reviewers (optional)
-->

<!--
## GIFs/Memes
- (optional)
-->

## Reviewer Checklist

This section is to be filled out by reviewers

### Testing

- [ ] This code was tested by somebody other than the developer. Do not merge until this has been done.


[MI-121]: https://lobsters.atlassian.net/browse/MI-121?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

juanfriss commented:
<img width="1065" alt="image" src="https://github.com/lob/ui-components/assets/83967528/5610c25b-a694-4072-9e51-752f9549e249">

when there's an error, it looks weird

kevinpjones commented:
> <img alt="image" width="1065" src="https://private-user-images.githubusercontent.com/83967528/316595984-5610c25b-a694-4072-9e51-752f9549e249.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzOTU2MDgsIm5iZiI6MTcxMTM5NTMwOCwicGF0aCI6Ii84Mzk2NzUyOC8zMTY1OTU5ODQtNTYxMGMyNWItYTY5NC00MDcyLTllNTEtNzUyZjk1NDllMjQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDE5MzUwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3ZmU0ZTRhZjU4ZmJjZTMxYTYzMGY4NGE2MjBlYTNhMzkyOGE5YjcyYjE4ODA4MjY3ZmFmNTVmZTBlNGJhZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tI2A5b-LDvayK17n3VbyOdMiplBlGBQXRZIWGEyHDzA">
> when there's an error, it looks weird

Good call.  Forgot to check the styles for the error state.  Fixed and added a test and story to cover that situation.

juanfriss commented on src/components/ImageFileUpload/__tests__/ImageFileUpload.spec.ts:
```diff
@@ -0,0 +1,189 @@
+
+    expect(fileInput).toBeInTheDocument();
+
+    const file = new File(['(⌐□_□)'], 'chucknorris.png', {
```
lol

juanfriss reviewed (approved):
there are couple of warnings which are easy to fix, other than that, looks good to me!
